packages/bff/src/routers/catalogRouter.ts

```diff
@@ -102,15 +102,43 @@ const catalogRouter = (
     )
     .delete(
       "/eservices/:eServiceId/descriptors/:descriptorId/documents/:documentId",
-      async (_req, res) => res.status(501).send()
+      async (req, res) => {
+        const ctx = fromBffAppContext(req.ctx, req.headers);
+        try {
+          await catalogService.deleteEServiceDocumentById(
+            unsafeBrandId(req.params.eServiceId),
+            unsafeBrandId(req.params.descriptorId),
+            unsafeBrandId(req.params.documentId),
+            ctx
```

Comment:
```suggestion
            ctx.headers
```
Since we don't need the whole ctx we can just pass the headers